compiler/luci-interpreter/src/kernels/StridedSlice.cpp

```diff
@@ -0,0 +1,150 @@
+}
+
+inline int32_t clampedIndex(int32_t index, int dim, bool pos_stride)
+{
+  return pos_stride
+             ? (index >= dim ? dim : positiveRemainder(std::min(std::max(index, -dim), dim), dim))
+             : (index < -dim ? -1
+                             : positiveRemainder(std::min(std::max(index, -dim), dim - 1), dim));
+}
+
+StridedSlice::StridedSlice(const Tensor *input, const Tensor *begin, const Tensor *end,
+                           const Tensor *strides, Tensor *output, const StridedSliceParams &params)
+    : KernelWithParams<StridedSliceParams>(params), _input(input), _begin(begin), _end(end),
+      _strides(strides), _output(output)
+{
+}
+
+void StridedSlice::configure()
+{
+  assert(_begin->shape().num_dims() == 1);
+  assert(_end->shape().num_dims() == 1);
+  assert(_strides->shape().num_dims() == 1);
+  assert(_input->element_type() == _output->element_type());
+  assert(_begin->element_type() == DataType::S32);
+  assert(_end->element_type() == DataType::S32);
+  assert(_strides->element_type() == DataType::S32);
+  assert(_input->shape().num_dims() <= kMaxDim);
+  assert(params().ellipsis_mask == 0);
+  assert(params().new_axis_mask == 0);
+  if (_input->element_type() == DataType::U8)
+  {
+    assert(_input->scale() == _output->scale());
+    assert(_input->zero_point() == _output->zero_point());
+  }
+  std::vector<int32_t> output_shape_vector;
+  for (int i = 0; i < _input->shape().num_dims(); i++)
+  {
+    int _idx = _input->shape().num_dims() - i - 1;
+    int32_t _stride_value = getTensorData<int32_t>(_strides)[_idx];
+    assert(_stride_value != 0);
+    bool _pos_stride = _stride_value > 0;
+    int _dim = _input->shape().dim(_idx);
+
+    int32_t _begin_value =
+        params().begin_mask & (1 << _idx)
+            ? _pos_stride ? 0 : _dim - 1
+            : clampedIndex(getTensorData<int32_t>(_begin)[_idx], _dim, _pos_stride);
+    int32_t _end_value = params().end_mask & (1 << _idx)
+                             ? _pos_stride ? 0 : _dim - 1
+                             : clampedIndex(getTensorData<int32_t>(_end)[_idx], _dim, _pos_stride);
+    bool _shrink_axis = params().shrink_axis_mask & (1 << _idx);
+    if (_shrink_axis)
+    {
+      _end_value = _begin_value + 1;
+    }
+    int32_t _dim_shape = std::ceil((_end_value - _begin_value) / static_cast<float>(_stride_value));
+    _dim_shape = _dim_shape < 0 ? 0 : _dim_shape;
+    if (!_shrink_axis)
+    {
+      output_shape_vector.push_back(_dim_shape);
+    }
+  }
+  Shape output_shape = Shape(output_shape_vector.size());
+  for (int i = 0; i < output_shape_vector.size(); i++)
+  {
+    output_shape.dim(i) = output_shape_vector[output_shape_vector.size() - i - 1];
+  }
+  _output->resize(output_shape);
+}
+
+void StridedSlice::execute() const
+{
+  std::vector<int32_t> starts;
+  std::vector<int32_t> stops;
+  std::vector<int32_t> strides;
+  for (int i = _input->shape().num_dims(); i < kMaxDim; i++)
```

Comment:
Where did this loop come from? I can't find similar code in TFLite.

Reply:
I referred from tensorflow v1.14 source and it changed on 2.1 version of tensorflow.
 If I should use another version of tensorflow, please tell me.
https://github.com/tensorflow/tensorflow/blob/00fad90125b18b80fe054de1055770cfb8fe4ba3/tensorflow/lite/kernels/strided_slice.cc#L190

Reply:
Oh, I found that it use a 2.1 version of tensorflow on Cmake file. I will update this kernel with that version. sorry for make confused.